src/pages/resultsView/querySummary/QuerySummary.tsx

```diff
@@ -39,24 +73,31 @@ export default class QuerySummary extends React.Component<{ routingStore:Extende
         return !!this.queryStore;
     }
 
-    private get singleStudyUI() {
-        return <div>
+    readonly singleStudyUI = remoteData({
+        await:()=>[
+            this.props.store.queriedStudies,
+            this.props.store.sampleLists,
+            this.props.store.samples,
+            this.props.store.patients,
+            this.props.store.genes
+        ],
+        invoke:()=>Promise.resolve(<div>
```

Comment:
oh boy.  JSX returned from invoke of remoteData?   I don't think I like this.   I think remoteData should return data

Reply:
why not? this way the caller only has to do myJSXElement.isComplete which is much more direct

Reply:
I would like to see a component which accepts mobxPromise[] and renders either success, pending or callback UI depending on status.   kid of extension of aggregate thing i did a while back

Reply:
but then the caller still has to know about all its dependencies and do more work. this enables encapsulating each thing with its dependencies so the caller doesnt need to know as much. I dont get the problem